mono/profiler/Makefile.am

```diff
@@ -100,7 +100,7 @@ endif
 if HAVE_STATIC_ZLIB
 zlib_dep = $(STATIC_ZLIB_PATH)
 else
-if HAVE_ZLIB
+if HAVE_SYS_ZLIB
 # The log profiler uses zlib for output compression when available.
 zlib_dep = -lz
 else
```

Comment:
`mono/profiler/log.c` currently checks `HAVE_SYS_ZLIB` so we need to add both the bundled sources here in the Makefile as well as including `"zlib.h"` when we have no system zlib.

Reply:
(and disable it for HOST_WASM in case we build the profiler there)

Reply:
I disagree on this point. All of log.c's lib consumption is gated behind HAVE_SYS_ZLIB. Adding new support for bundled or static zlib to the log profiler is out of scope for this PR.